src/main.c

```diff
@@ -69,22 +72,24 @@ int main(int argc, char *argv[]) {
    RSCompress compress;
    RSBufferCircle circle;
    RSOutput output = {0};
+   RSAudio audio;
    rsConfigLoad(&config);
    // We have to call this one first since it might change the config width/height
    rsXlibSystemCreate(&system, &config);
    rsCompressCreate(&compress, &config);
    size_t capacity = (size_t)(config.duration * config.framerate);
    rsBufferCircleCreate(&circle, capacity);
-
+   rsAudioCreate(&audio, &config);
+   uint8_t *rawSamples = rsMemoryCreate((size_t)audio.data.size);
```

Comment:
One allocation per program launch... same can be done with the buffer of frames (don't have to call malloc(30 x 60 x ptr_size) every time we save video.. but didn't want to change that in this PR, out of the scope. Also the samples are copied with the wraparound at once to this buffer, so that when grabbing samples to encode into aac we don't have to use the slower get, as all the samples are already in the correct order in this buffer.

Reply:
The reason I make a copy is so it can keep recording while saving without overwriting the data its trying to save. You should probably do something similar here.

Reply:
Oh right, I'll change that then

Reply:
that's done